### PR TITLE
Garden functionality

### PR DESCRIPTION
**Pull request by cmpprg**
### Type of change made:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Styling (Styling to new or existing files)
- [ ] Testing
### Detailed Description
Added Endpoint to api that retrieves all plants from garden. Hooked up api to MyGarden page. Mapped through results and rendered GardenPlant components. Wrapped Garden in a scrollview. Modified GardenPlant to dynamically display information.

### Why is this change required? What problem does it solve?
Takes information from API and renders to the garden, Core functionality of App.

### Was any previously written code changed?
Transformed MyGarden from functional component to a class component to store state. 

### Directory modified:
- ./components/pages/MyGarden
- ./components/GardenPlant
- ./api/plantAPI

### CODE REVIEW

File: components/GardenPlant.js [hunk 1]
```diff
@@ -1,12 +1,13 @@
 import React from 'react';
-import { View, Text, TouchableHighlight} from 'react-native';
+import { View, Text, TouchableOpacity} from 'react-native';
```
Comment (kimalajoy) [inline components/GardenPlant.js:2]: I think this was a good change over from TouchableHighlight

File: components/pages/MyGarden.js [hunk 2]
```diff
@@ -1,25 +1,50 @@
-import React from 'react';
-import { View, Text, ImageBackground, SafeAreaView} from 'react-native';
+import React, { Component } from 'react';
+import { View, Text, ImageBackground, SafeAreaView, ScrollView} from 'react-native';
```
Comment (kimalajoy) [inline components/pages/MyGarden.js:2]: Getting that scroll view in there so we can see all the little plant babies!

**Review by kimalajoy — APPROVED** [review]
This looks great. I am glad we got this functionality running and I thought you made some thoughtful changes with the scrollview and touchableopacity

File: components/GardenPlant.js (hunk 1 above)
Comment (cmpprg) [reply components/GardenPlant.js:2]: Ya, I think it looks better. We should change the one on the Search page as well.

File: components/pages/MyGarden.js (hunk 2 above)
Comment (cmpprg) [reply components/pages/MyGarden.js:2]: It was surprisingly easy to implement.

**Pull request merged**